### DISCUSSION TITLE
improve dijkstra

### DISCUSSION THREAD

**Discussion by lengyijun** [petgraph/petgraph#698] [General]
https://arxiv.org/pdf/2311.11793#page=41.35

 According to Appendix B, we need to implement a modified fibonacci heap to enable Universal Optimality

**Comment by starovoid** [comment]
Wouldn't such an implementation have too large a runtime constant? The Fibonacci heap is famous for its overhead. But I agree that on sufficiently large graphs, the Fibonacci/Brodal/Binomial/etc heaps approach starts to win performance. You can add support for Fibonacci and any other heap as an alternative to `BinaryHeap`, specifying the type of priority queue used as generic. But I think this will be possible no earlier than the release of version `0.7`
